zoo_service.py

```diff
@@ -0,0 +1,107 @@
+from flask import Flask, request, abort
+from flask_sqlalchemy import SQLAlchemy
+from flask_marshmallow import Marshmallow
+from marshmallow import fields, ValidationError
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = 'mysql+mysqlconnector://myk_user:fwQH-totF8Pv@localhost:3306/lab6'
+db = SQLAlchemy(app)
+ma = Marshmallow(app)
+
+
+class Fish(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    weight_in_kg = db.Column(db.Float)
+    thermoregulation = db.Column(db.String(15))
+    lifetime_years = db.Column(db.Float)
+    animal_type = db.Column(db.String(50))
+    required_aquarium_capacity_liters = db.Column(db.Float)
+    required_temperature = db.Column(db.Integer)
+    required_lighting_level = db.Column(db.String(20))
+
+    def __init__(
+            self, weight_in_kg, thermoregulation, lifetime_years, animal_type,
+            required_aquarium_capacity_liters, required_temperature, required_lighting_level
+    ):
+        self.weight_in_kg = weight_in_kg
+        self.thermoregulation = thermoregulation
+        self.lifetime_years = lifetime_years
+        self.animal_type = animal_type
+        self.required_aquarium_capacity_liters = required_aquarium_capacity_liters
+        self.required_temperature = required_temperature
+        self.required_lighting_level = required_lighting_level
+
+
+class FishSchema(ma.Schema):
+    id = fields.Int()
+    weight_in_kg = fields.Float()
+    thermoregulation = fields.String()
+    lifetime_years = fields.Float()
+    animal_type = fields.String()
+    required_aquarium_capacity_liters = fields.Integer()
+    required_temperature = fields.Integer()
+    required_lighting_level = fields.String()
+
+
+fish_schema = FishSchema(exclude=["id"])
+fishes_schema = FishSchema(only=["id", "animal_type", "lifetime_years"], many=True)
```

Comment:
```suggestion
FISH_SCHEMA = FishSchema(exclude=["id"])
FISH_LIST_SCHEMA = FishSchema(only=["id", "animal_type", "lifetime_years"], many=True)
```

All caps to identify global constants. Better naming to clearly distinguish between single-object and list schemas.

Reply:
Changed. Also renamed `app`, `db` and `ma`.